dbt/adapters/spark/impl.py

```diff
@@ -142,24 +147,15 @@ def list_relations_without_caching(
                 return []
 
         relations = []
-        for row in results:
-            if len(row) != 4:
-                raise dbt.exceptions.RuntimeException(
-                    f'Invalid value from "show table extended ...", '
-                    f'got {len(row)} values, expected 4'
-                )
-            _schema, name, _, information = row
-            rel_type = RelationType.View \
-                if 'Type: VIEW' in information else RelationType.Table
-            is_delta = 'Provider: delta' in information
-            is_hudi = 'Provider: hudi' in information
+        view_names = views.columns["viewName"].values()
+
+        for tbl in tables:
+            rel_type = RelationType('view' if tbl['tableName'] in view_names else 'table')
+            _schema = tbl['namespace'] if 'namespace' in tbl else tbl['database']
             relation = self.Relation.create(
                 schema=_schema,
-                identifier=name,
+                identifier=tbl['tableName'],
                 type=rel_type,
-                information=information,
-                is_delta=is_delta,
-                is_hudi=is_hudi,
```

Comment:
Given that we won't be returning this information from the cache anymore (totally reasonable), let's update places where we expected to access this information, and ensure it will be doing a sensible thing without the information:
1. Method `get_columns_in_relation` checked if `information` was present
2. Macro `spark__alter_relation_add_remove_columns` and table + seed materializations checked `is_delta` / `is_hudi`

In the latter case, the goal is to determine whether the file format of the actual table matches the configured file format that the user has provided. We have two choices:
- Trust the user, and just use `config.get('file_format')` instead of checking. This would only raise issues if someone runs a table `using parquet`, then switches it to Delta, and dbt attempts `create or replace`.
- Run an extra query (`describe table`) to determine the actual file format, and use that information to update the relation that each materialization will cache

Reply:
@jtcohen6 I think it makes more sense to run an extra query (`describe table`) to get extra information when it is needed, and the code is more robust. But I've found there isn't a good way to update the cached relation. Do you have any better suggestions for updating a cached relation?

Reply:
@TalkWIthKeyboard You're right, it feels like we're missing an `update_relation` method in the core adapter cache ([`cache.py`](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/adapters/cache.py)). It would work a lot like [`add`](https://github.com/dbt-labs/dbt-core/blob/3996a69861d5ba9a460092c93b7e08d8e2a63f88/core/dbt/adapters/cache.py#L316-L328), but instead of [`setdefault`](https://github.com/dbt-labs/dbt-core/blob/3996a69861d5ba9a460092c93b7e08d8e2a63f88/core/dbt/adapters/cache.py#L251) (return the existing cached relation if it already exists, insert if it doesn't), it should always overwrite the existing cache entry for that relation.

That might have some performance implications, since the cache is shared across thread-safe, such that any writes require holding a lock. It would be worth testing with concurrency.

If you're up for it, it would be possible to continue developing that in this same repo / PR, since the cache is an attribute of the adapter class:
```python
# dbt/adapters/spark/impl.py

SparkRelationsCache(RelationsCache):
    def update_relation:
        ...

SparkAdapter(SQLAdapter):
    cache = SparkRelationsCache()
```

If we can get it working, the right move would then be to move that code upstream into the core cache logic.

Reply:
Sure, I am happy to continue developing. Thanks for your advice.